### PR TITLE
Fixes 2969:  fetch (POST) snapshots for content_templates

### PR DESCRIPTION
**Pull request by Andrewgdewar**
## Summary

Adds the following API: 
POST /repositories/snapshots/for_date/

Body
`{     repository_uuids: ['abcde'],   date: “2023-01-01” }`

Response: 
```
[
   {
      repository_uuid: "abcde"   
      is_after: true,
      match: { SNAPHOT_JSON    }     
   }  
]
```

## Testing steps

- Create a few repositories, wait for them to snapshot. 
- Hit the above endpoint using your selected tool, filling in the repository_uuids and targeted time (UTC).

Additional information: 
- If the date of the nearest snap is greater than 24 hours past the specified date the "is_true" flag will be false.
- If no match is found, the "is_after" defaults to false.
- The Date will be compared as 24 hours inclusive. IE, if you snapshotted yesterday, and then now, and set the "Date" to yesterday's date, you'd likely still return that snap as a match without the "is_after" flag being switched. 
- Currently I am not surfacing not_found errors if the user has some matches, ie if you had 3 repository_uuids, and only had snaps for the first 2, the third (regardless of cause of not found) will just return an empty "match" and false "is_after" 

## Checklist

- [x] Tested with snapshotting feature disabled and pulp server URL not configured if appropriate

### CODE REVIEW

**Comment by jlsherrill** [comment]
https://issues.redhat.com/browse/HMS-2969

File: pkg/dao/snapshots.go [hunk 1]
```diff
@@ -236,6 +237,112 @@ func (sDao *snapshotDaoImpl) FetchLatestSnapshot(repoConfigUUID string) (api.Sna
...
+				).First(&snapshot).Error
+
+			if err != nil {
+				return []api.ListSnapshotByDateResponse{}, err
+			}
+
+			if snapshot.Base.UUID != "" {
+				listResponse[i].IsAfter = true
+			} else {
+				continue
```
[the remark this answers is not in the document]
Comment (Andrewgdewar) [reply pkg/dao/snapshots.go:331]: Most of my struggle on this ticket was fighting with gorms methods to get what I wanted out of them.

[the remark this answers is not in the document]
Comment (Andrewgdewar) [reply pkg/dao/snapshots.go:337]: Okay after looking at this query, what would happen in the case that there are 1000 snapshots. and I want the one that was 500 (relative to creation date). 

Would I be expected to loop over all of them?

[the remark this answers is not in the document]
Comment (Andrewgdewar) [reply pkg/dao/snapshots.go:301]: Okay, I can't get a single query to work at all. 

Although I've been able to get a bunch of queries to run against pgAdmin, the "Raw" function isn't working with the above query, and more advanced queries Plpgsql throw an exception (something about not allowing a dynamic values in preset queries or some such). 

If you can find a work-around please let me know.

**Review by jlsherrill — APPROVED** [review]

**Pull request merged**